src/test/java/core/basesyntax/service/parser/FruitRecordDtoParserImplTest.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax.service.parser;
+
+import core.basesyntax.dto.FruitRecordDto;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.fileservice.FileReader;
+import core.basesyntax.service.fileservice.FileReaderForCsvImpl;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class FruitRecordDtoParserImplTest {
+    private static final String validCsvFile = "src/test/resources/valid_instructions.csv";
+    private static final String firstInvalidCsvFile = "src/test/resources/invalid_instructions.csv";
+    private static final String secondInvalidCsvFile =
+            "src/test/resources/invalid_instructions_2.csv";
+    private static final String thirdInvalidCsvFile =
+            "src/test/resources/invalid_instructions_3.csv";
+    private static final String fourthInvalidCsvFile =
+            "src/test/resources/invalid_instructions_4.csv";
+    private static final String emptyCsvFile = "src/test/resources/empty.csv";
+    private static final FruitRecordDtoParser parser = new FruitRecordDtoParserImpl();
+    private static final FileReader fileReader = new FileReaderForCsvImpl();
```

Comment:
Constant names use CONSTANT_CASE: all uppercase letters, with each word separated from the next by a single underscore.

Reply:
it's a initial solution, haven't done refactoring yet)